src/engraving/dom/lyricsline.cpp

```diff
@@ -142,13 +152,183 @@ LyricsLineSegment::LyricsLineSegment(LyricsLine* sp, System* parent)
     setGenerated(true);
 }
 
+LyricsLineSegment::LyricsLineSegment(const ElementType& type, LyricsLine* sp, System* parent, ElementFlags f)
+    : LineSegment(type, sp, parent, f)
+{
+    setGenerated(true);
+}
+
 double LyricsLineSegment::baseLineShift() const
 {
     if (lyricsLine()->isEndMelisma()) {
         return -0.5 * absoluteFromSpatium(lineWidth());
     }
 
-    Lyrics* lyrics = lyricsLine()->lyrics();
+    Lyrics* segLyrics = lyrics();
+    return -style().styleD(Sid::lyricsDashYposRatio) * segLyrics->fontMetrics().xHeight();
+}
+
+//=========================================================
+//   PartialLyricsLine
+//=========================================================
+PartialLyricsLine::PartialLyricsLine(EngravingItem* parent)
+    : LyricsLine(ElementType::PARTIAL_LYRICSLINE, parent)
+{
+    setGenerated(false);
+}
+
+PartialLyricsLine::PartialLyricsLine(const PartialLyricsLine& other)
+    : LyricsLine(other)
+{
+    m_isEndMelisma = other.m_isEndMelisma;
+}
+
+LineSegment* PartialLyricsLine::createLineSegment(System* parent)
+{
+    PartialLyricsLineSegment* seg = new PartialLyricsLineSegment(this, parent);
+    seg->setTrack(track());
+    seg->setColor(color());
+    return seg;
+}
+
+PropertyValue PartialLyricsLine::getProperty(Pid propertyId) const
+{
+    switch (propertyId) {
+    case Pid::VERSE:
```

Comment:
Why does PartialLyricsLine need to specifically override the VERSE property? It loooks to me like the parent class should return the same thing? Same question goes for setProperty and propertyDefault

Reply:
The parent `LyricsLine` class doesn't have a verse property - it's taken from the line's "parent"/associated lyric.  Because partial lyric lines won't always have an associated lyric the line needs to store this info itself.

Reply:
Oh right! 👍